### PR TITLE
xds: integrate backend metric API to client load reporting

### PR DESCRIPTION
**Pull request by voidzcy**

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/ClientLoadCounter.java [hunk 1]
```diff
@@ -72,6 +85,13 @@ void recordCallFinished(Status status) {
     }
   }
 
+  @Override
+  void recordMetric(String name, double value) {
+    MetricRecorder recorder =
+        metricRecorders[(int) (Thread.currentThread().getId() % THREAD_BALANCING_FACTOR)];
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/ClientLoadCounter.java]: Is this documented in some spec?

Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/ClientLoadCounter.java]: No, this is a dirty trick we came up with to deal with atomic update of two items while keeping thread contention not too severe. But, it's an implementation detail. :smirk:

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/ClientLoadCounter.java]: Casting a long value to int could overflow and get a negative result.
```
System.out.println((int) 2147483700l  % 64);
```

```
-12
```

Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/ClientLoadCounter.java]: The number to be casted will be less than `THREAD_BALANCING_FACTOR` after the modulo operation (more precisely, it's remainder operation). So overflow should not happen.

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/ClientLoadCounter.java]: Oh, I overlooked there is a bracket surrounding modulo expression. Is `Thread.getId()` never negative?

Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/ClientLoadCounter.java]: Yes.
> [The thread ID is a positive long number generated when this thread was created.](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#getId())

**Pull request merged**